core/services/functions/listener.go

```diff
@@ -121,7 +121,6 @@ type FunctionsListener struct {
 	mbOracleEvents    *utils.Mailbox[log.Broadcast]
 	serviceContext    context.Context
 	serviceCancel     context.CancelFunc
-	chStop            chan struct{}
```

Comment:
Alternatively, you could upgrade this to a `utils.StopChan` and drop both context fields (since `context.Context` [isn't supposed to be stored](https://pkg.go.dev/context))
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Reply:
They are meant to be request-scoped and static tools have expectations about it being passed as a func arg.
This approach has created problems within this project where we end up with two contexts, one from background and one from the request call, and then there is no clear way to merge them coherently. There is some new `package context` API coming soon that makes this a bit easier to manage when it happens, but one or the other is really just used for the cancellation signal, and the unnecessary baggage of `context.Context` creates the issue.